decorators/decorators.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+import functools
+import sys
+
+
+def get_fqn(obj):
+    """
+    This function tries to determine the fully qualified name (FQN) of the callable that is provided.
+    It only works for classes, methods and functions.
+    It is unable to properly determine the FQN of class instances, static methods and class methods.
+    """
+    path = [getattr(obj, '__module__', None)]
+    if sys.version_info[0] > 2:
```

Comment:
Why doing the Python version check at all? We don't need that: if `obj` has the attribute `__qualname__ ` then use it, otherwise go for the other strategy (based on `im_class`).